sdk/core/core-client/test/serviceClient.spec.ts

```diff
@@ -162,6 +162,33 @@ describe("ServiceClient", function () {
       assert.deepEqual(request!.headers.get("authorization"), "Bearer testToken");
     });
 
+    it("should use endpoint to build scope", async function () {
```

Comment:
@joheredi To make sure it's backward compatible I keep the test here https://github.com/Azure/azure-sdk-for-js/pull/21125/files#diff-6789bb88ffa3f4fcfc260721566bf7267615e8dbc0f7d0d1eadf3a220284fe38R138-R163 and add a new one for endpoint.